src/library_gl.js

```diff
@@ -1069,52 +1073,74 @@ var LibraryGL = {
 
   glCompressedTexImage2D__sig: 'viiiiiiii',
   glCompressedTexImage2D: function(target, level, internalFormat, width, height, border, imageSize, data) {
-#if ASSERTIONS
-    assert(GL.currentContext.compressionExt);
```

Comment:
The functions `glCompressedTexImage2D` and `glCompressedTexSubImage2D` are core WebGL 1 functions. They don't need an extension to work. Also, the variable `compressionExt` was present when the extension `WEBGL_compressed_texture_s3tc` was enabled, but that extension is not a requirement to call the `glCompressedTex*` functions.

We could validate that

    if (internalFormat == GL_COMPRESSED_RGBA_S3TC_DXT1_EXT) assert(GL.currentContext.s3tcCompressionExt);
    else if (internalFormat == GL_COMPRESSED_RGBA_PVRTC_4BPPV1_IMG) assert(GL.currentContext.pvrtcCompressionExt);

and so on, but that is wrong since GL should generate a GL error on failure and never abort, and redundant, since it is the browser's job already to validate the format and generate a GL error when the function is called with an unsupported compression type, so we'd just be duplicating internal browser validation work.